gobblin-yarn/src/main/java/gobblin/yarn/YarnService.java

```diff
@@ -94,6 +99,7 @@
   private final Config config;
 
   private final EventBus eventBus;
+  private final EventSubmitter eventSubmitter;
```

Comment:
Also we need to make this an `Optional`, which will only be present if `ConfigurationKeys.METRICS_ENABLED_KEY` is set to `true`. This allows us to control whether to emit events or not.